Added Required Queries

JaiPannu-IITI opened the pull request:
### Overview

This PR is necessary for https://github.com/PalisadoesFoundation/talawa-admin/pull/3838

Parent Issue: #3230 

Please merge together.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced an option for users to view organizations they created.
  - Enhanced organization queries with an optional case-insensitive name filter for improved search functionality.
  - Added filtering capability for organizations where users are members, allowing for case-insensitive searches.

- **Refactor**
  - Improved query handling and result transformation for organization listings and memberships, ensuring more responsive and intuitive filtering.

- **Bug Fixes**
  - Enhanced error handling for user authentication and query execution failures.

- **Tests**
  - Added new test suites for validating the functionality of organization queries and created organizations, covering various scenarios including authentication and error handling.
  - Introduced tests for the `resolveOrganizationsWhereMember` function to ensure proper error handling and filtering.
  - Added tests for the `resolveCreatedOrganizations` function to validate its behavior across multiple scenarios.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

JaiPannu-IITI commented:
@palisadoes PR is ready for review

(coderabbit parsing error solved in https://github.com/PalisadoesFoundation/talawa-api/pull/3356)

mpdesai90 commented:
@pranshugupta54, @GlenDsza, Please review this PR.

Cioppolo14 commented:
@JaiPannu-IITI Please fix the coderabbit issue, so we can see what comments it has for this PR.

JaiPannu-IITI commented:
@Cioppolo14 Please add ignore sensitive file label for coderabbit fix.

JaiPannu-IITI commented on src/graphql/types/User/createdOrganizations.ts:
```diff
@@ -0,0 +1,32 @@
+							),
+						limit: 20,
+					});
+				} catch (error) {
+					console.error("Error fetching created organizations:", error);
+					throw new Error(
+						"Failed to retrieve organizations created by the user",
+					);
+				}
```
already done

JaiPannu-IITI commented:
@Cioppolo14 Patch Test don't make sense here since modified files don't have associated test files written, and writing them  for this type is itself a big issue which is out of scope for chat development. Please ignore that test for now.

GlenDsza commented:
> @Cioppolo14 Patch Test don't make sense here since modified files don't have associated test files written, and writing them for this type is itself a big issue which is out of scope for chat development. Please ignore that test for now.

@JaiPannu-IITI Please add tests to covered the modified lines as well. Covering tests in separate PRs wont be ideal. It would be better to add new features along with testcases

JaiPannu-IITI commented:
@GlenDsza What I am trying to explain is that these files are not having any spec file associated with them right now (they are at 0% coverage) , that means issues are raised already and assigned to people to write test for these files, I only made changes which were necessary to make admin panel work right now. I want to work on fixing real time chat but pipeline is unreachable, these queries are required to solve a admin panel blocker that would help me to start working on chat.

palisadoes commented:
1. Please add the tests. Our policy is to make sure all our PR checks pass to the greatest extent possible.
2. I recognize that you are fixing a blocking issue, but we need to maintain the integrity of the code base

JaiPannu-IITI commented:
alright sir

JaiPannu-IITI commented:
Not ready yet, will add more test cases

JaiPannu-IITI commented:
@coderabbitai can you write more tests for test/graphql/types/User/organizationsWhereMember.test.ts

JaiPannu-IITI commented:
@palisadoes PR is ready for review 

Summary:

- Added Required queries to fix organizations in admin and use debounced search on them.
- Wrote tests for created files 

<img width="1062" alt="Screenshot 2025-03-14 at 2 39 22 PM" src="https://github.com/user-attachments/assets/8ec0b83e-f806-4793-ad7b-fd0aaec3c1b2" />
<img width="1375" alt="Screenshot 2025-03-14 at 12 52 24 PM" src="https://github.com/user-attachments/assets/f9f40f8f-75c0-4fb7-bc1a-92780557de63" />

- Refactored tests folder to similar hierarchies as original files providing easy navigation to new contributors.
